README.md

```diff
@@ -284,10 +318,27 @@ Mysql2::Client.new(
   :sslcapath => '/path/to/cacerts',
   :sslcipher => 'DHE-RSA-AES256-SHA',
   :sslverify => true, # Removed in MySQL 8.0
-  :ssl_mode = :disabled / :preferred / :required / :verify_ca / :verify_identity,
+  :ssl_mode => :disabled / :preferred / :required / :verify_ca / :verify_identity,
   )
 ```
 
+For MySQL versions 5.7.11 and higher, use `:ssl_mode` to prefer or require an
+SSL connection and certificate validation. For earlier versions of MySQL, use
+the `:sslverify` boolean. For details on each of the `:ssl_mode` options, see
+[https://dev.mysql.com/doc/refman/8.0/en/connection-options.html](https://dev.mysql.com/doc/refman/8.0/en/connection-options.html#option_general_ssl-mode).
+
+The `:ssl_mode` option will also set the appropriate MariaDB connection flags:
+
+| `:ssl_mode`        | MariaDB option value                 |
+| ---                | ---                                  |
+| `:disabled`        | MYSQL_OPT_SSL_ENFORCE = 0            |
+| `:required`        | MYSQL_OPT_SSL_ENFORCE = 1            |
+| `:verify_identity` | MYSQL_OPT_SSL_VERIFY_SERVER_CERT = 1 |
+
+MariaDB does not support the `:preferred` or `:verify_ca` options. For more information SSL/TLS in MariaDB, see
+[https://mariadb.com/kb/en/securing-connections-for-client-and-server/](https://mariadb.com/kb/en/securing-connections-for-client-and-server/)
+and [https://mariadb.com/kb/en/mysql_optionsv/#tls-options](https://mariadb.com/kb/en/mysql_optionsv/#tls-options)
```

Comment:
We are printing the values of the ssl_mode in the warning message below. Maybe it's helpful to have a document about the mapping table of ssl_mode and the value or column for that on this table, such as `:disabled` | `0` | `MYSQL_OPT_SSL_ENFORCE = 0`

```
rb_warn("Your mysql client library does not support ssl_mode %d", val);
```

Reply:
I think the awkward message is really because the translation from Ruby atom to C constant is in Ruby code while the mode setting action is in C code. The C code has the error but doesn't know what the original Ruby atom was.

A deeper cleanup might be moving the whole method into Ruby space, but I think it would be better get in the way of good enough for now.

Reply:
OK. I agree with it!